### PR TITLE
[webflux] Enable testcontainer support for MariaDB too

### PR DESCRIPTION
**Pull request by gzsombor**
and add converter from BitSet to Boolean which is needed by the MariaDB driver

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/files.js [hunk 1]
```diff
@@ -1410,7 +1410,7 @@ const serverFiles = {
             templates: [{ file: 'testcontainers/mariadb/my.cnf', method: 'copy', noEjs: true }],
         },
         {
-            condition: generator => ['mysql', 'postgresql', 'mssql'].includes(generator.prodDatabaseType) && generator.reactive,
+            condition: generator => ['mysql', 'postgresql', 'mssql', 'mariadb'].includes(generator.prodDatabaseType) && generator.reactive,
```
Comment (mshima) [inline generators/server/files.js:1413]:
```suggestion
            condition: generator.reactiveSqlTestContainers,
```

File: generators/generator-base.js [hunk 2]
```diff
@@ -2284,6 +2284,7 @@ module.exports = class extends PrivateBase {
         dest.searchEngine = config.searchEngine;
         dest.cacheProvider = config.cacheProvider;
         dest.enableHibernateCache = config.enableHibernateCache;
+        dest.reactiveSqlTestContainers = config.reactiveSqlTestContainers;
```
Comment (mshima) [inline generators/generator-base.js:2287]: Can be dropped, this would load the config if it's stored at .yo-rc.json
```suggestion
```

Comment (gzsombor) [reply generators/generator-base.js:2287]: If I remove it, the reactiveSqlTestContainers variable is not accessible from the templates.

File: generators/server/files.js (hunk 1 above)
Comment (gzsombor) [reply generators/server/files.js:1413]: Thx, I forgot to update it

File: generators/generator-base.js (hunk 2 above)
Comment (mshima) [reply generators/generator-base.js:2287]: Ok, you are using to load into entity generator.
`reactiveSqlTestContainers` doesn't exists in `config`, you need to calculate it.
Updated the suggestion.

File: generators/server/index.js [hunk 3]
```diff
@@ -272,6 +272,9 @@ module.exports = class extends BaseBlueprintGenerator {
                     } else {
                         this.containerVersion = 'latest';
                     }
+                    this.reactiveSqlTestContainers =
+                        this.jhipsterConfig.reactive &&
+                        ['mysql', 'postgresql', 'mssql', 'mariadb'].includes(this.jhipsterConfig.prodDatabaseType);
```
Comment (mshima) [inline generators/server/index.js:277]: Can be removed, it's already loaded by loadAppConfig().
```suggestion
```

**Comment by gzsombor** [comment]
Great, everything, but the webflux tests were run.

**Comment by mshima** [comment]
Please change 
https://github.com/jhipster/generator-jhipster/blob/161519958d55863962abc03a8907ab24fc05ffbb/.github/workflows/webflux.yml#L48
to
```
            (contains(github.event.head_commit.message, '[webflux]') ||
                contains(github.event.pull_request.title, '[webflux]')) &&
```

**Comment by MathieuAA** [comment]
I'm okay with `reactiveSqlTestContainers` as it's not a function (nor a function call) but a variable.

**Comment by avdev4j** [comment]
As I remember we struggle with Webflux builds because the reactive option was totally broken (problem with compilation or backend tests...). That's why we had to merge the feature before having a fix. We did not want to wait more and avoid as much as possible conflicts (a little bit hard with branch with more than 100 commits :( ).

Does @nassimerrahoui or @adilabed any of you can help on this? I think we should open a specific PR to fix those tests.

**Comment by mshima** [comment]
I think /account/password and /account/settings errors are related to the backend:
Eg: maybe https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/java/package/service/InvalidPasswordException.java.ejs returns error 401
while https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/java/package/web/rest/errors/InvalidPasswordException.java.ejs returns error 400.

The others seems to be cypress failing to login or keeping the user logged.

Anyway the errors should be fixed with spring 2.3 directly.

**Comment by gzsombor** [comment]
Yes, as these change only touch the reactive-sql codepaths, it should not effect the MongoDB tests, or the oauth2 ones, I'm going to merge it, so I can finalize the reactive-entity support changes!

**Pull request merged**